utils/mstfwmanager/Makefile

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (C) 2024
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=mstfwmanager
+PKG_VERSION:=4.26.0
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/Mellanox/mstflint
+PKG_MIRROR_HASH:=9ab90387f295e974e3e6f6e8b0ce7e5f37749e0b1a6977c1b5c2ee6c2c2b80f6
+PKG_SOURCE_VERSION:=ff014c1e873e86255c07a396d0f3588ad018cc2a
+PKG_SOURCE_DATE:=2023-11-08
+
+PKG_MAINTAINER:=Til Kaiser <mail@tk154.de>
+PKG_LICENSE:=GPL-2.0-or-later
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_FIXUP:=autoreconf
+PKG_BUILD_PARALLEL:=1
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/mstfwmanager
+  SECTION:=Utilities
+  CATEGORY:=Utilities
+  TITLE:=Mellanox firmware update and query utility
+  DEPENDS:=@TARGET_x86 \
+    +libcurl +liblzma +libopenssl +libstdcpp +libxml2 +zlib
+endef
+
+define Package/mstfwmanager/description
+    Mellanox firmware update and query utility which scans 
+    the system for available Mellanox devices (only mst
+    PCI devices) and performs the necessary firmware updates.
+endef
+
+CONFIGURE_ARGS += \
+	--enable-fw-mgr \
+	--disable-inband
+
+TARGET_CFLAGS += \
+	-D_GNU_SOURCE
```

Comment:
This is still required?

Reply:
I believe so. The package accesses u_int8_t, u_int16_t, and u_int32_t which are protected by _GNU_SOURCE in musl libc <sys/types.h>: https://git.musl-libc.org/cgit/musl/tree/include/sys/types.h#n59